projects/pgai/pgai/vectorizer/vectorizer.py

```diff
@@ -461,18 +482,51 @@ async def run(self) -> int:
         res = 0
         loops = 0
 
-        async with await psycopg.AsyncConnection.connect(self.db_url) as conn:
+        async with await psycopg.AsyncConnection.connect(
+            self.db_url, autocommit=True
+        ) as conn:
             await register_vector_async(conn)
             await self.vectorizer.config.embedding.setup()
             while True:
-                if not self._continue_processing(loops, res):
+                if not await self._continue_processing(conn, loops, res):
                     return res
                 items_processed = await self._do_batch(conn)
                 if items_processed == 0:
                     return res
                 res += items_processed
                 loops += 1
 
+    async def _continue_processing(
+        self, conn: AsyncConnection, loops: int, res: int
+    ) -> bool:
+        """Determine whether to continue processing based on vectorizer status
+        and custom logic.
+
+        Args:
+            conn (AsyncConnection): Database connection for checking vectorizer status.
+            loops (int): Number of processing loops completed.
+            res (int): Result from the previous processing iteration.
+
+        Returns:
+            bool: True if processing should continue, False otherwise.
+
+        Raises:
+            Exception: If the vectorizer row is not found in the database.
+        """
+        if self.features.disable_vectorizers:
+            async with conn.cursor() as cursor:
+                await cursor.execute(
+                    self.queries.is_vectorizer_disabled_query,
+                    [self.vectorizer.id],
+                )
+                row = await cursor.fetchone()
+                if not row:
+                    raise Exception("vectorizer row not found")
+                if row[0]:
+                    return False
```

Comment:
Shouldn't this return the value of `disabled` from the row in the database rather than a hardcoded `False`?

Reply:
No because if it's not disable we still need to check the `self._continue_processing_hook` to see if we should continue processing.

The hook on cloud is a function that calculates if we are near the 80% execution time that we are allowed to run.

Reply:
Ahh! gotcha.